### ISSUE TITLE
No unneeded ternary check?

### ISSUE THREAD

**Issue by dingo-d** [WordPress/WordPress-Coding-Standards#1291]
When using ternary can we have a check that will see if the ternary is needed or not? [ESLint](https://eslint.org/docs/rules/no-unneeded-ternary) uses this rule. This would be thrown when we want to store boolean in a variable. For instance

```php
<?php

$var = 'a';
// Example of an unneeded ternary:
$bla = $var === 'a' ? true : false;

// Can be written as:
$bla = $var === 'a';
```

Don't know if we have this or not (since I'm not getting warnings I'm guessing we don't have it).

Also, I've stumbled into this: https://www.drupal.org/project/drupal/issues/1838368

Would this be useful to have as a rule?

**Comment by jrfnl** [comment]
> Also, I've stumbled into this: https://www.drupal.org/project/drupal/issues/1838368

As stated in various comments in that thread, the memory leak issue which that ticket refers to was fixed in PHP 5.4, so IMO is quite irrelevant.

**Comment by dingo-d** [comment]
We'll move this to the PHPCSExtra ruleset in the future: https://github.com/PHPCSStandards/PHPCSExtra/issues/157
